cnxarchive/utils/ident_hash.py

```diff
@@ -56,19 +58,17 @@ def split_legacy_hash(legacy_hash):
 
 def split_ident_hash(ident_hash, split_version=False):
     """Returns a valid id and version from the <id>@<version> hash syntax."""
-    if HASH_CHAR not in ident_hash:
-        ident_hash = '{}@'.format(ident_hash)
     split_value = ident_hash.split(HASH_CHAR)
-    if split_value[0] == '':
-        raise ValueError("Missing values")
 
-    try:
-        id, version = split_value
-    except ValueError:
+    if len(split_value) > 2 or '' in split_value:
         raise IdentHashSyntaxError(ident_hash)
 
-    # Validate the id.
+    if len(split_value) == 1:
+        id, version = split_value[0], ''
```

Comment:
I think we raise `IdentHashShortId` first so we can get the uuid and then if the version is also missing, we can get the version from the uuid. :smile: